packages/edit-site/src/components/global-styles/variations-panel.js

```diff
@@ -2,16 +2,26 @@
  * WordPress dependencies
  */
 import { store as blocksStore } from '@wordpress/blocks';
+import { privateApis as blockEditorPrivateApis } from '@wordpress/block-editor';
 import { useSelect } from '@wordpress/data';
 import { __experimentalItemGroup as ItemGroup } from '@wordpress/components';
+
 /**
  * Internal dependencies
  */
-
 import { NavigationButtonAsItem } from './navigation-button';
+import { unlock } from '../../lock-unlock';
+
+const { useGlobalStyle } = unlock( blockEditorPrivateApis );
 
-function getCoreBlockStyles( blockStyles ) {
-	return blockStyles?.filter( ( style ) => style.source === 'block' );
+// Only core block styles (source === block) or block styles with
+// a matching theme.json style variation will be configurable via
+// Global Styles.
+function getFilteredBlockStyles( blockStyles, variations ) {
```

Comment:
This function now opens up Global Styles to allow for non-core block style variations to be configured through the site editor. 

At this stage, inner block type and element styles for the block style variation are not editable however this could be worked on in a follow-up.